Source/Application/Network/AbbotsfordRestaurantListProvider.swift

```diff
@@ -0,0 +1,87 @@
+import Foundation
+
+struct AbbotsfordRestaurantListProvider: RestaurantListProviding, EndpointClient {
+
+    static let baseURLString: String = "https://developers.zomato.com/api/v2.1/"
+
+    static let queryItems: [URLQueryItem] = [
+        URLQueryItem(name: "entity_id", value: "98284"),
+        URLQueryItem(name: "entity_type", value: "subzone"),
+        URLQueryItem(name: "count", value: "10"),
+    ]
+
+    static let headerItems: [URLHeaderItem] = [
+        URLHeaderItem(field: "user-key", value: "95ab57dfb9c5ebe2bab89c4af8a58062"),
+    ]
+
+    private let requestServicing: RequestServicing
+
+    init(requestServicing: RequestServicing) {
+        self.requestServicing = requestServicing
+    }
+
+    enum Endpoint: String, EndpointRepresenting {
+
+        case locationDetails = "location_details"
+
+        var path: String {
+            return self.rawValue
+        }
+    }
+
+    func withRestaurantList(execute closure: @escaping (Result<[Restaurant], AnyError>) -> Void) {
+
+        let url: URL! = URL(string: self.urlString(for: .locationDetails))
+        let handler: (Result<NetworkResponse, NetworkError>) -> Void = { result in
+            switch result {
+            case let .success(response):
+                var requestResponse: RequestResponse?
+                let decoder = JSONDecoder()
+                decoder.keyDecodingStrategy = .convertFromSnakeCase
+                do {
+                    requestResponse = try decoder.decode(RequestResponse.self, from: response.data)
+                } catch {
+                    closure(.failure(AnyError(error)))
```

Comment:
The 'if let' after this 'catch' statement may not execute after this 'catch' is run because the 'request response' would always be 'nil', but I think that with the way that this is written, we leave space for there to be future developer error where we introduce code into the seams between the 'catch' and the 'if let'.

```swift
catch {
  closure(.failure(...))
}
// Code may enter here, unknowingly creating a new state in our supposed finite state machine.
if let requestResponse = requestResponse {
  closure(.success(requestResponse.extractRestaurants()))
}
```

I think explicitly carving out the possible results (in states, or transitions) here would help mitigate those kinds of future side-effecting bugs.

Do you think that this is something we can tackle now?